framework/PostProcessors/ETimporter.py

```diff
@@ -0,0 +1,554 @@
+
+#External Modules---------------------------------------------------------------
+import numpy as np
+import xml.etree.ElementTree as ET
+import copy
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+from .PostProcessor import PostProcessor
+from utils import InputData
+from utils import xmlUtils as xmlU
+import Files
+import Runners
+#Internal Modules End-----------------------------------------------------------
+
+
+class ETImporter(PostProcessor):
+  """
+    This is the base class of the postprocessor that imports Event-Trees (ETs) into RAVEN as a PointSet
+  """
+  def __init__(self, messageHandler):
+    """
+      Constructor
+      @ In, messageHandler, MessageHandler, message handler object
+      @ Out, None
+    """
+    PostProcessor.__init__(self, messageHandler)
+    self.printTag = 'POSTPROCESSOR ET IMPORTER'
+    self.ETFormat = None
+    self.allowedFormats = ['OpenPSA']
+
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(ETImporter, cls).getInputSpecification()
+    inputSpecification.addSub(InputData.parameterInputFactory("fileFormat", contentType=InputData.StringType))
+
+    return inputSpecification
+
+
+  def initialize(self, runInfo, inputs, initDict) :
+    """
+      Method to initialize the pp.
+      @ In, runInfo, dict, dictionary of run info (e.g. working dir, etc)
+      @ In, inputs, list, list of inputs
+      @ In, initDict, dict, dictionary with initialization options
+      @ Out, None
+    """
+    PostProcessor.initialize(self, runInfo, inputs, initDict)
+
+  def _localReadMoreXML(self, xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs got
+      @ In, xmlNode, xml.etree.Element, Xml element node
+      @ Out, None
+    """
+
+    paramInput = ETImporter.getInputSpecification()()
+    paramInput.parseNode(xmlNode)
+    self._handleInput(paramInput)
+
+  def _handleInput(self, paramInput):
+      fileFormat = paramInput.findFirst('fileFormat')
+      self.fileFormat = fileFormat.value
+      if self.fileFormat not in self.allowedFormats:
+          self.raiseAnError(IOError,
+                            'ETImporterPostProcessor Post-Processor ' + self.name + ', format ' + str(self.fileFormat) + ' : is not supported')
+
+  def run(self, inputs):
+    """
+      This method executes the postprocessor action.
+      @ In,  inputs, list, list of file objects
+      @ Out, None
+    """
+    self.outputDict = self.runOpenPSA(inputs)
```

Comment:
I think we should avoid changing the state variable self.outputDict, because this not thread safe. Please use local variable. see https://github.com/idaholab/raven/wiki/Avoid-Using-Model-State-when-in-Parallel-Execution-Context